crates/rpc/rpc/src/trace.rs

```diff
@@ -549,6 +568,33 @@ where
         }
         traces
     }
+
+    /// Returns all storage slots accessed during transaction execution along with their access
+    /// counts.
+    pub async fn trace_transaction_storage_access(
+        &self,
+        tx_hash: B256,
+    ) -> Result<Option<TransactionStorageAccess>, Eth::Error> {
```

Comment:
we're mainly interested in this info across the entire block, so this should take a blockid and we want to trace the entire block like:

https://github.com/paradigmxyz/reth/blob/1e0045bc4cbf3baa068cfd0ff91276017a2dafb8/crates/rpc/rpc/src/trace.rs#L455-L477